components/brave_ads/browser/ads_notification_handler.cc

```diff
@@ -46,10 +56,20 @@ void AdsNotificationHandler::OnClose(
     const std::string& id,
     bool by_user,
     base::OnceClosure completed_closure) {
+
+  base::OnceClosure completed_closure_local = base::BindOnce(&AdsNotificationHandler::OperationCompleted,
+      base::Unretained(this), id);
+  pending_callbacks_.emplace(id, std::move(completed_closure));
+
+#if defined (OS_ANDROID)
+  last_dismissed_notification_id_ = id;
+  no_ui_shutdown_timer_.Stop();
+#endif
+
```

Comment:
If `ads_service_` is `null`, will this code end up being called twice (due to the `if` clause below here that calls `OnClose` again)? Wasn't sure, but maybe something to check...

Reply:
Fixed